MarkBind/markbind
# Add support for generating a timestamp

**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**
• [X] Enhancement to an existing feature


Resolves #187 



**What is the rationale for this request?**
User would like an additional pre-defined variable within MarkBind to insert timestamps

**What changes did you make? (Give an overview)**
- Add a new Util function `initPageTimestamp(page)` that overrides value with the 'timestamp' key within `this.userDefinedVariablesMap[base]` with the current time.
- Function is called in `generatePages` and `regenerateAffectedPages`
- Update UserGuide [here](https://deploy-preview-222--markbind.netlify.com/userguide/contentauthoring#use-markbind-variables)
**Provide some examples**

Used in the same way as glyphicons | 
--- | 
![feature-timestamp](https://user-images.githubusercontent.com/31084833/40220157-bd743edc-5aaa-11e8-92d8-e22cf4f1c891.PNG) |

Known limitations |
--- |
Clicking the `MarkBind` link when the timestamp is placed in a component header will not work |
Color of link and panel type="primary" blends together as shown above |

**Is there anything you'd like reviewers to focus on?**
- code variable naming / style

**Testing instructions:**
1. Insert {{timestamp}} anywhere in the source,
1. Check it renders correctly
1. Click on the hyperlink within the generated timestamp and check it works.
